components/playlist/browser/resources/components/playerSeeker.tsx

```diff
@@ -0,0 +1,253 @@
+    border-radius: calc(var(--progress-bar-height) * 2);
+  }
+`
+
+function formatTime (time: number) {
+  const stringFormat = (t: number) => String(t).padStart(2, '0')
+  const hours = stringFormat(Math.floor(time / 3600))
+  const minutes = stringFormat(Math.floor((time % 3600) / 60))
+  const seconds = stringFormat(Math.floor(time % 60))
+  return `${hours}:${minutes}:${seconds}`
+}
+
+class DragController {
+  constructor (
+    videoRef: React.RefObject<HTMLVideoElement>,
+    progressElem: HTMLProgressElement,
+    updateProgressValue: (value: number) => void
+  ) {
+    this.#videoRef = videoRef
+    this.#progressElem = progressElem
+    this.#updateProgressValue = updateProgressValue
+  }
+
+  #videoRef: React.RefObject<HTMLVideoElement>
+  #progressElem: HTMLProgressElement
+  #updateProgressValue: (value: number) => void
+
+  #isMouseDown = false
+  #isDragging = false
+
+  // Indicate the state of the video when mouse was down.
+  #initialState?: {
+    playing: boolean
+    clientX: number
+    value: number
+  } = undefined
+
+  interpolatePxToValue = (clientX: number) => {
+    // clientWidth : clientX = max : newValue
+    return (clientX / this.#progressElem.clientWidth) * this.#progressElem.max
+  }
+
+  onMouseDown = (event: MouseEvent) => {
+    if (!this.#videoRef.current) {
+      return
+    }
+
+    this.#isMouseDown = true
+    this.#initialState = {
+      playing: !this.#videoRef.current?.paused,
+      clientX: event.clientX,
+      value: this.#progressElem.value
+    }
+
+    // Pause video during drag-and-drop session.
+    this.#videoRef.current.pause()
+
+    this.#videoRef.current.currentTime = this.interpolatePxToValue(
+      this.#initialState.clientX - this.#progressElem.offsetLeft
+    )
+
+    document.addEventListener('mousemove', this.onMouseMove)
+    document.addEventListener('mouseup', this.onMouseUp)
+  }
+
+  onMouseMove = (event: MouseEvent) => {
+    if (!this.#videoRef.current) {
+      return
+    }
+
+    if (!this.#isMouseDown) {
+      return
+    }
+
+    const dragThreshold = 5
+    if (
+      !this.#isDragging &&
+      Math.abs(event.clientX - this.#initialState!.clientX) > dragThreshold
+    ) {
+      this.#isDragging = true
+    }
+
+    if (!this.#isDragging) {
+      return
+    }
+
+    // Update value
+    const newTime = this.interpolatePxToValue(
+      event.clientX - this.#progressElem.offsetLeft
+    )
+
+    const oldTime = this.#videoRef.current.currentTime
+    this.#videoRef.current.currentTime = newTime
+
+    if (oldTime === this.#videoRef.current.currentTime) {
+      // In case the newTime has changed by time smaller than a second,
+      // the |currentTime| won't be updated. In this case <progress> could feel
+      // like jank. In order to avoid that, force to update the progress ui.
+      this.#updateProgressValue(newTime)
+    }
+  }
+
+  onMouseUp = (event: MouseEvent) => {
+    document.removeEventListener('mousemove', this.onMouseMove)
```

Comment:
Just double checking, but what happens if you:
1. Start dragging
2. Move the mouse outside the contents area
3. Release the mouse
4. Move the mouse back into the contents area

Reply:
Nah, sounds perfect to me!